ComponentKit/Core/Scope/CKComponentScopeHandle.mm

```diff
@@ -92,6 +92,16 @@ - (instancetype)newHandleWithStateUpdates:(const CKComponentStateUpdateMap &)sta
                                                controller:_controller];
 }
 
+- (instancetype)newHandleWillBeReacquiredDueToScopeCollision
```

Comment:
Instead of actually reusing the same component scope handle I decided to create an identical handle that hasn't already been acquired.